app/domain/services/create_course/service.rb

```diff
@@ -0,0 +1,39 @@
+class Services::CreateCourse::Service
+  def process(course_uuid:, ecosystem_uuid:)
+    return nil if course_uuid.nil? or ecosystem_uuid.nil?
+
+    start_time     = Time.now
+    start_time_str = start_time.utc.iso8601(6)
+
+    ecosystem = Ecosystem.find_by uuid: ecosystem_uuid
+
+    fail Errors::AppRequestValidationError.new("Ecosystem #{ecosystem_uuid} does not exist. Course cannot be created.") \
```

Comment:
This should probably be a `AppUnprocessableError`, since the schema validation must have passed to get to this point.